### PR TITLE
native: add support test on native posix mode

### PR DESCRIPTION
**Pull request by LingaoM**
Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

### CODE REVIEW

**Comment by michal-narajowski** [comment]
@LingaoM Could you update the PR and resolve conflicts?

**Comment by LingaoM** [comment]
> @LingaoM Could you update the PR and resolve conflicts?

Yes, it may be postponed to next Monday.

File: autoptsclient_common.py [hunk 1]
```diff
@@ -1221,7 +1228,7 @@ def check_args(self, args):
             board_power(args.ykush, True)
             time.sleep(1)
 
-        if 'tty_file' in args:
+        if args.tty_file:
```
Comment (LingaoM) [inline autoptsclient_common.py:1231]: It looks more like a bug before. When we use QEMU, we will report an error(Nonetype)

**Comment by LingaoM** [comment]
> @LingaoM Could you update the PR and resolve conflicts?

@michal-narajowski Updated.

File: ptsprojects/zephyr/iutctl.py [hunk 2]
```diff
@@ -386,18 +408,14 @@ def init_stub():
     ZEPHYR = ZephyrCtlStub()
 
 
-def init(kernel_image, tty_file, board=None, use_rtt2pty=False):
+def init(args):
```
Comment (LingaoM) [inline ptsprojects/zephyr/iutctl.py:411]: The reason for the change here is that it is more concise.

**Comment by mkasenberg** [comment]
I'm running some tests with https://github.com/zephyrproject-rtos/zephyr/pull/36584. Feedback coming soon ;)

**Comment by LingaoM** [comment]
> I'm running some tests with https://github.com/zephyrproject-rtos/zephyr/pull/36584. Feedback coming soon ;)

It is highly recommended that you open another terminal: sudo btmon

File: autoptsclient_common.py (hunk 1 above)
Comment (mkasenberg) [reply autoptsclient_common.py:1231]: Oh, I came here too late and I made you some conflict here with https://github.com/intel/auto-pts/pull/479.

Comment (mkasenberg) [reply autoptsclient_common.py:1231]: It was truly a bug. Just 'if args.tty_file:' wasn't enough though. tty_file is not always in that args.

Comment (LingaoM) [reply autoptsclient_common.py:1231]: I will rebase...

**Comment by mkasenberg** [comment]
Works great!

File: autoptsclient_common.py (hunk 1 above)
Comment (LingaoM) [reply autoptsclient_common.py]: @mkasenberg Updated.

**Review by mkasenberg — APPROVED** [review]

**Comment by LingaoM** [comment]
@michal-narajowski Please review, thanks!

**Review by michal-narajowski — APPROVED** [review]
LGTM

**Pull request merged**